components/gitpod-db/src/typeorm/team-db-impl.ts

```diff
@@ -203,6 +203,19 @@ export class TeamDBImpl implements TeamDB {
             throw new ResponseError(ErrorCodes.NOT_FOUND, "A team with this ID could not be found");
         }
         const membershipRepo = await this.getMembershipRepo();
+
+        if (role != "owner") {
+            const ownerCountAfterChange = await membershipRepo.count({
+                teamId,
+                userId: Not(userId),
+                role: "owner",
+                deleted: false,
+            });
+            if (ownerCountAfterChange < 1) {
+                throw new ResponseError(ErrorCodes.CONFLICT, "Team must retain at least one owner");
+            }
```

Comment:
Interesting -- I've never seen typeorm's `Not` being used. Minor suggestion: Maybe it could be a bit simpler and less "magical" to just check the total number of owners? 😇 

```suggestion
            const ownerCount = await membershipRepo.count({
                teamId,
                role: "owner",
                deleted: false,
            });
            if (ownerCount <= 1) {
                throw new ResponseError(ErrorCodes.CONFLICT, "Team must retain at least one owner");
            }
```

Reply:
Yep, this also works due to the `if (role == 'owner')` check. We could drop that check and do the Not exclusion but not sure if we gain much - it's just a bit more resilient to role spec changes.

Reply:
Used your suggestion.